src/reader/decompress.rs

```diff
@@ -3,13 +3,62 @@
 
 use std::io::Read;
 
-use arrow::datatypes::ToByteSlice;
 use bytes::{Bytes, BytesMut};
 use fallible_streaming_iterator::FallibleStreamingIterator;
 use snafu::ResultExt;
 
 use crate::error::{self, Error};
-use crate::proto::CompressionKind;
+use crate::proto::{self, CompressionKind};
+
+#[derive(Clone, Copy, Debug)]
+pub struct Compression {
+    compression_type: CompressionType,
+    /// No compression chunk will decompress to larger than this size.
+    /// Use to size the scratch buffer appropriately.
+    max_decompressed_block_size: usize,
+}
+
+impl Compression {
+    pub fn from_proto(
+        kind: proto::CompressionKind,
+        compression_block_size: Option<u64>,
+    ) -> Option<Self> {
+        // Spec states default is 256K
+        let max_decompressed_block_size = compression_block_size.unwrap_or(256 * 1024) as usize;
```

Comment:
nit: Consider moving `256 * 1024` into a const.

Reply:
Good point, moved to const